confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSource.java

```diff
@@ -102,32 +99,25 @@ private Schema inferSchema(FailureCollector failureCollector) {
       options.put("basic.auth.user.info", conf.getSchemaRegistryApiKey() + ':' + conf.getSchemaRegistryApiSecret());
       CachedSchemaRegistryClient schemaRegistryClient =
         new CachedSchemaRegistryClient(conf.getSchemaRegistryUrl(), 2, options);
-      Schema initialSchema = conf.getSchema(failureCollector);
       List<Schema.Field> newFields = new ArrayList<>();
       boolean keySchemaShouldBeAdded = conf.getKeyField() != null;
-      boolean messageSchemaShouldBeAdded = conf.getValueField() != null;
-      for (Schema.Field field : initialSchema.getFields()) {
-        if (field.getName().equals(conf.getKeyField())) {
-          Schema keySchema = fetchSchema(schemaRegistryClient, conf.getTopic() + "-key");
-          newFields.add(Schema.Field.of(field.getName(), keySchema));
-          keySchemaShouldBeAdded = false;
-        } else if (field.getName().equals(conf.getValueField())) {
-          Schema valueSchema = fetchSchema(schemaRegistryClient, conf.getTopic() + "-value");
-          newFields.add(Schema.Field.of(field.getName(), valueSchema));
-          messageSchemaShouldBeAdded = false;
-        } else {
-          newFields.add(field);
-        }
-      }
+      boolean messageSchemaShouldBeAdded = conf.getvalueSchema() != null;
```

Comment:
Is it only used once?

Reply:
the variable `messageSchemaShouldBeAdded`?